### ISSUE TITLE
Rule Change: no-duplicate-imports should report for duplicate re-export all

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#21087]
### What rule do you want to change?

no-duplicate-imports

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

Other

### Example code

```js
export { a } from "mod";

export { b } from "mod";    // This is detected
export * from "mod";        // This is not detected as duplicate
```

### What does the rule currently do for this code?

The rule reports for the named re-export `export { b } from "mod"` when `options.includeExports = true`.
But it does not report for when using a re-export all (`export * from "..."`).
In this case the exports are the same when removing the named re-exports and keeping the re-export all.
This seems to be an oversight as other combinations (e.g. named re-export and namespaced re-export all is tested).

### What will the rule do after it's changed?

Extend the option `includeExports` to also report for the case of re-exports with the same source and:
- one named re-export (without rename) and one non-namespaced re-export all
- two re-export alls (either with or without namespace)

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I would see this as change as a bugfix and not a breaking change.

**Comment by yuvrajkarna2717** [comment]
Hi! I'd like to work on this issue.

The proposed behavior makes sense to me, and I'd be happy to implement the changes to `no-duplicate-imports` (including handling duplicate `export * from` and mixed named/export-all re-exports when `includeExports` is enabled), along with the necessary tests.

Could you please assign this issue to me? Thanks!

**Comment by mdjermanovic** [comment]
> * two re-export alls (either with or without namespace)

Can you provide some examples for this?

**Comment by DMartens** [comment]
The three combinations for re-export alls are:
```javascript
// Both without namespace
export * from "mod";
export * from "mod"; // This export is useless as the members are already exported

// One with and one without namespace
export * from "mod";
export * as mod from "mod"; // Highly unlikely that user wanted to export the same members multiple times

// Both with namespaces
export * as a from "mod";
export * as b from "mod"; // Same as above
```

**Comment by mdjermanovic** [comment]
> ```js
> // Both without namespace
> export * from "mod";
> export * from "mod"; // This export is useless as the members are already exported
> ```

This is already reported: https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZHVwbGljYXRlLWltcG9ydHM6IFtcImVycm9yXCIsIHsgXCJpbmNsdWRlRXhwb3J0c1wiOiB0cnVlIH1dKi9cblxuLy8gQm90aCB3aXRob3V0IG5hbWVzcGFjZVxuZXhwb3J0ICogZnJvbSBcIm1vZFwiO1xuZXhwb3J0ICogZnJvbSBcIm1vZFwiOyAvLyBUaGlzIGV4cG9ydCBpcyB1c2VsZXNzIGFzIHRoZSBtZW1iZXJzIGFyZSBhbHJlYWR5IGV4cG9ydGVkIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

> ```js
> // One with and one without namespace
> export * from "mod";
> export * as mod from "mod"; // Highly unlikely that user wanted to export the same members multiple times
> ```

I don't think this falls under "all imports from a single module that can be merged exist in a single import statement", which is what this rule enforces (https://eslint.org/docs/latest/rules/no-duplicate-imports#rule-details), because these two cannot be merged in a single import or export.

> ```js
> // Both with namespaces
> export * as a from "mod";
> export * as b from "mod"; // Same as above
> ```

This is also already reported:
https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZHVwbGljYXRlLWltcG9ydHM6IFtcImVycm9yXCIsIHsgXCJpbmNsdWRlRXhwb3J0c1wiOiB0cnVlIH1dKi9cblxuLy8gQm90aCB3aXRoIG5hbWVzcGFjZXNcbmV4cG9ydCAqIGFzIGEgZnJvbSBcIm1vZFwiO1xuZXhwb3J0ICogYXMgYiBmcm9tIFwibW9kXCI7IC8vIFNhbWUgYXMgYWJvdmUiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

**Comment by DMartens** [comment]
I agree that this change does not follow this description but strictly speaking the third case (both with a namespace) also does not match it.
It is not possible to recreate separate namespaces (in the example `a` and `b`) with a single export.

**Comment by mdjermanovic** [comment]
The third case can be rewritten this way to use only one import from the module:

```js
import * as mod from "mod";

export { mod as a, mod as b };
```

**Comment by DMartens** [comment]
Didn't think about using an `import` for this case.
As the change does not match the rule's intent, I am going to close this issue.
